### PR TITLE
chore: run go test in ci workflow

### PR DESCRIPTION
**Pull request by eddycharly**
This PR runs `go test` in CI workflow.

### CODE REVIEW

**Comment by eddycharly** [comment]
Anything I should do to avoid this ?
image

**Comment by apelisse** [comment]
You'd need to be an org member or something? I can look that up right after.

**Comment by eddycharly** [comment]
I think i'm a k/k member, maybe it's not enough ?

**Comment by eddycharly** [comment]
Now test fail 🤔

**Comment by apelisse** [comment]
That seems completely unrelated? is that a flake?

**Comment by eddycharly** [comment]
Not failing locally, checking.

**Comment by eddycharly** [comment]
Ahah, i had a cluster running locally, once i deleted it test started to fail.

**Comment by apelisse** [comment]
Ohhh, that's fishy, the tests should definitely not depend on your running environment ...

**Comment by eddycharly** [comment]
> Ohhh, that's fishy, the tests should definitely not depend on your running environment ...

```
					openapiclient.NewLocalFiles(c.localFilesDir), // satisfy expectation users' expectation that provided local files are used
					openapiclient.NewLocalCRDFiles(c.localFilesDir),
```

Is this expected that we use the same dir for those two clients in the chain ?
First one is supposed to load openapi schema files and second is supposed to load CRD definitions.

**Comment by apelisse** [comment]
Yeah, I think it's loading all the files and then decides if they are openapi or CRDs (from which we extract the openapi). That should be fine, WDYT?

**Comment by eddycharly** [comment]
My 2 cents is we could provide two flags, nothing prevents those two flags to point at the same dir.

**Comment by eddycharly** [comment]
And if I apply the fix in https://github.com/kubernetes-sigs/kubectl-validate/pull/26 test panics 🤔

**Comment by apelisse** [comment]
Go for it :-)

**Comment by apelisse** [comment]
Let's see if @alexzielenski can help?

**Comment by eddycharly** [comment]
Sure, from what I understand, tests are going to depend on the kubernetes version flag ? (because validation and message can vary slightly)

**Comment by apelisse** [comment]
> Anything I should do to avoid this ? image

Your name shows "contributor", not "member", that's probably why. If you're a k/k member, you almost certainly can be a kubernetes-sigs member too, especially with the help you give us here.

**Comment by alexzielenski** [comment]
Since the tests are making use of the `RootCommand` they get `openapiclient.NewKubeConfig(c.kubeConfigOverrides),` added automatically. We should add a way to disable that in tests to avoid running clusters being used by tests.

File: testcases/manifests/error_quantity.yaml [hunk 1]
```diff
@@ -1,19 +1,14 @@
 # {
 #   "metadata": {},
 #   "status": "Failure",
-#   "message": "Pod.core \"my-pod\" is invalid: [<nil>: Invalid value: \"\": \"spec.containers[0].resources.requests.cpu\" must validate one and only one schema (oneOf). Found none valid, spec.containers[0].resources.requests.cpu: Invalid value: \"boolean\": spec.containers[0].resources.requests.cpu in body must be of type string: \"boolean\"]",
```
[the remark this answers is not in the document]
Comment (eddycharly) [reply testcases/manifests/error_quantity.yaml:4]: Ok, I will revert this change once #30 merges

[the remark this answers is not in the document]
Comment (alexzielenski) [reply testcases/manifests/error_quantity.yaml:4]: it just merged

[the remark this answers is not in the document]
Comment (eddycharly) [reply testcases/manifests/error_quantity.yaml]: done

**Pull request merged**